src/Panels/ConnectedPlayersPanel.cs

```diff
@@ -0,0 +1,146 @@
+
+        public override void Start()
+        {
+            _playerLabels = new List<UILabel>();
+            _kickButtons = new List<UIButton>();
+
+            // Activates the dragging of the window
+            AddUIComponent(typeof(UIDragHandle));
+
+            backgroundSprite = "GenericPanel";
+            color = new Color32(110, 110, 110, 250);
+
+            // Grab the view for calculating width and height of game
+            UIView view = UIView.GetAView();
+
+            // Center this window in the game
+            relativePosition = new Vector3(view.fixedWidth / 2.0f - 180.0f, view.fixedHeight / 2.0f - 240.0f);
+
+            width = 360;
+            height = 445;
+
+            // Title Label
+            UILabel title = this.CreateTitleLabel("Connected Players", new Vector2(80, -20));
+
+            // Close this dialog
+            _closeButton = this.CreateButton("Close", new Vector2(10, -375));
+            _closeButton.eventClick += (component, param) =>
+            {
+                isVisible = false;
+            };
+
+            base.Start();
+        }
+
+        public override void Update()
+        {
+            // Update the list of players and the kick buttons
+            if (isVisible && _playerListLastHash != MultiplayerManager.Instance.PlayerList.GetHashCode())
+            {
+                _playerListLastHash = MultiplayerManager.Instance.PlayerList.GetHashCode();
+
+                foreach (UILabel label in _playerLabels)
+                {
+                    label.Hide();
+                    label.Disable();
+                }
+
+                int topOffset = -75;
+                int currentPlayerOffset = 0;
+                int playerOffset = -30;
+
+                // Enable Host to see and kick all players
+                if (MultiplayerManager.Instance.CurrentRole == MultiplayerRole.Server)
+                {
+                    foreach (UIButton button in _kickButtons)
+                    {
+                        button.Hide();
+                        button.Disable();
+                    }
+
+                    List<string> players = MultiplayerManager.Instance.PlayerList.ToList();
+
+                    // List all the players
+                    for (int i = 0; i < players.Count; i++)
+                    {
+                        string player = players[i];
+
+                        if (i < _playerLabels.Count)
+                        {
+                            _playerLabels[i].text = player;
+                            _playerLabels[i].Show();
+                            _playerLabels[i].Enable();
+
+                            if (player != MultiplayerManager.Instance.CurrentServer.HostPlayer.Username)
+                            {
+                                _kickButtons[i].Show();
+                                _kickButtons[i].Enable();
+                            }
+
+                            _kickButtons[i].eventClick += (component, param) =>
```

Comment:
The kick button only works when the player connects for the second time because this handler is not registered in the else case below. Also this handler is added multiple times.
You could simplify this whole loop by not reusing the elements of _playerLabels and _kickButtons and just regenerating the list completely.